patches/chrome-installer-mini_installer-mini_installer.cc.patch

```diff
@@ -11,7 +11,133 @@ index 1f39abbeb2579098a54fa82612199f0307ea03b4..be08f89c43599fc766048d230572e71a
  // Opens the Google Update ClientState key. If |binaries| is false, opens the
  // key for Google Chrome or Chrome SxS (canary). If |binaries| is true and an
  // existing multi-install Chrome is being updated, opens the key for the
-@@ -873,7 +873,7 @@ ProcessExitResult WMain(HMODULE module) {
+@@ -142,6 +142,105 @@ void SetInstallerFlags(const Configuration& configuration) {
+ }
+ #endif  // GOOGLE_CHROME_BUILD
+ 
++#if defined(BRAVE_CHROMIUM_BUILD)
++typedef StackString<128> ReferralCodeString;
++
++bool ParseStandardReferralCode(const wchar_t* filename, ReferralCodeString* referral_code) {
++  // Scan backwards for last dash in filename.
++  const wchar_t* scan = filename + lstrlen(filename) - 1;
++  while (scan != filename && *scan != L'-')
++    --scan;
++
++  if (*scan++ != L'-')
++    return false;
++
++  // Standard referral code is 6 characters.
++  const wchar_t* ref_code = scan;
++  if (lstrlen(ref_code) != 6)
++    return false;
++
++  // Ensure that first half of referral code is alphabetic.
++  for (int i = 0; i < 3; ++i) {
++    if ((ref_code[i] < L'a' || ref_code[i] > L'z') &&
++        (ref_code[i] < L'A' || ref_code[i] > L'Z'))
++      return false;
++  }
++
++  // Ensure that second half of referral code is numeric.
++  for (int i = 3; i < 6; ++i) {
++    if (ref_code[i] < L'0' || ref_code[i] > L'9')
++      return false;
++  }
++
++  if (!referral_code->assign(ref_code))
++    return false;
++
++  return true;
++}
++
++bool ParseExtendedReferralCode(const wchar_t* filename, ReferralCodeString* referral_code) {
++  // Scan backwards for second-to-last dash in filename, since this
++  // type of referral code has an embedded dash.
++  const wchar_t* scan = filename + lstrlen(filename) - 1;
++  while (scan != filename && *scan != L'-')
++    --scan;
++
++  if (*scan-- != L'-')
++    return false;
++
++  while (scan != filename && *scan != L'-')
++    --scan;
++
++  if (*scan++ != L'-')
++    return false;
++
++  // Ensure that referral code is alphabetic.
++  const wchar_t* ref_code = scan;
++  int dashes = 0;
++  for (int i = 0; i < lstrlen(ref_code); ++i) {
++    if ((ref_code[i] < L'a' || ref_code[i] > L'z') &&
++        (ref_code[i] < L'A' || ref_code[i] > L'Z') && (ref_code[i] != L'-'))
++      return NULL;
++    if (ref_code[i] == L'-')
++      ++dashes;
++  }
++
++  // Ensure that referral code contains exactly one dash.
++  if (dashes != 1)
++    return false;
++
++  if (!referral_code->assign(ref_code))
++    return false;
++
++  return true;
++}
++
++bool ParseReferralCode(const wchar_t* program, ReferralCodeString* referral_code) {
```

Comment:
nitpick: should `referral_code` be a reference instead of a pointer? (ex: `ReferralCodeString& referral_code`). Kind of nice IMO, because references can't be set to null / reseated

Reply:
That seems reasonable to me, I can make that change.